File: Ruby/acm_quiz_001.rb
def quiz3Nplus1(n, sum)
    if n <= 1
        return sum
    elsif n % 2 == 1
        return quiz3Nplus1(n*3+1, sum + 1)
    else
        return quiz3Nplus1(n/2, sum + 1)
    end
end

mMax = 0
(1..10).each do |q|
    tmp = quiz3Nplus1(q,1)
    if tmp > mMax
        mMax = tmp
    end
end
puts "max len : #{mMax}"

File: JavaScript/AcmQuiz001.js
function quiz3Nplus1(n, sum) {
    if (n <= 1) {
        return sum;
    } else if (n%2 == 1) {
        return quiz3Nplus1(n*3+1, ++sum);
    } else {
        return quiz3Nplus1(n/2, ++sum);
    }
}

function main() {
    var max = 0;
    for (var i = 1 ; i < 10 ; i++) {
        var tmp = quiz3Nplus1(i, 1);
        if (tmp > max) {
            max = tmp;
        }
    }
    console.log("max len : " + max);
}

main();
